pkg/kv/kvserver/rangefeed/processor_test.go

```diff
@@ -634,6 +635,45 @@ func TestProcessorMemoryBudgetReleased(t *testing.T) {
 	})
 }
 
+// TestIntentScannerOnError tests that when a processor is given with an intent
+// scanner constructor that fails to create a scanner, the processor will fail
+// to start gracefully. Regression for #127204.
+func TestIntentScannerOnError(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	t.Helper()
+	stopper := stop.NewStopper()
+	cfg := testConfig{
+		Config: Config{
+			RangeID: 2,
+			Stopper: stopper,
+			Span:    roachpb.RSpan{Key: roachpb.RKey("a"), EndKey: roachpb.RKey("z")},
+			Metrics: NewMetrics(),
+		},
+	}
+	sch := NewScheduler(SchedulerConfig{
+		Workers:         1,
+		PriorityWorkers: 1,
+		Metrics:         NewSchedulerMetrics(time.Second),
+	})
+	require.NoError(t, sch.Start(context.Background(), stopper))
+	cfg.Scheduler = sch
+	s := NewProcessor(cfg.Config)
+	require.ErrorContains(t, s.Start(stopper, func() (IntentScanner, error) {
+		return nil, errors.New("scanner error")
+	}), "scanner error")
+	ctx := context.Background()
```

Comment:
I'd move this to near the top of the function and use it in the other places you've also used context.Background()

Reply:
Done.